src/lib/types.ts

```diff
@@ -83,32 +83,37 @@ export abstract class PostgrestBuilder<T> implements PromiseLike<PostgrestRespon
       body: JSON.stringify(this.body),
     })
       .then(async (res) => {
-        let error, data, count
+        let error = null
+        let data = null
+        let count = null
+
         if (res.ok) {
-          error = null
-          if (this.method !== 'HEAD') {
-            const isReturnMinimal = this.headers['Prefer']?.split(',').includes('return=minimal')
-            data = isReturnMinimal ? null : await res.json()
-          } else {
-            data = null
+          const isReturnMinimal = this.headers['Prefer']?.split(',').includes('return=minimal')
+          if (this.method !== 'HEAD' && !isReturnMinimal) {
+            const text = await res.text()
+            if (text && text !== '') {
+              try {
+                data = JSON.parse(text)
+              } catch (_) {
+                error = { message: 'Failed to parse json response' }
+              }
+            }
```

Comment:
Maybe we should just do nothing in catch? Not sure if it's a good idea to "forge" PostgREST errors. Otherwise you'll also need other fields in PostgrestError:

https://github.com/supabase/postgrest-js/blob/9ecb9851429bb15b6569ad7dfd24b7363e335e93/src/lib/types.ts#L8-L13

Also could do without the if methinks

Reply:
Because we always expect json response from PostgREST. So if the response.text has content and we can't parse it to json. An error message is reasonable.

Actually, the error message is for us. I don't think users can do anything with this error. It's either PostgREST bug or postgrest-js bug.

Reply:
Agreed, either way the user gets an unexpected error and files a bug. I guess we can keep old behavior and add special case for `text === ''`? Loud and explicit error > silent error IMO.

Reply:
let do less. I remove try/catch block.